pagerduty/event_orchestration_cache_variable.go

```diff
@@ -0,0 +1,119 @@
+package pagerduty
+
+import (
+	"context"
+	"fmt"
+)
+
+type EventOrchestrationCacheVariableService service
+
+type EventOrchestrationCacheVariableCondition struct {
+	// A PCL string: https://developer.pagerduty.com/docs/ZG9jOjM1NTE0MDc0-pcl-overview
+	Expression string `json:"expression,omitempty"`
+}
+
+// Configuration for a cache variable changes depending on the type:
+//   - if `Type` is `recent_value`; then use `Regex` and `Source`
+//   - if `Type` is `trigger_event_count`; then use `TTLSeconds`
+type EventOrchestrationCacheVariableConfiguration struct {
+	Type       string `json:"type,omitempty"`
+	Regex      string `json:"regex,omitempty"`
+	Source     string `json:"source,omitempty"`
+	TTLSeconds int    `json:"ttl_seconds,omitempty"`
+}
+
+// A reference to a related object (e.g. an User, etc)
+type EventOrchestrationCacheVariableReference struct {
```

Comment:
Please prefer to extend `resourceReference` like the rest of resource references in [here](https://github.com/heimweh/go-pagerduty/blob/bfc8dce0a3ffc87473867b75b81c52cb9edd9d9e/pagerduty/references.go#L3).

Reply:
Addressed in https://github.com/heimweh/go-pagerduty/pull/149/commits/cb1dc594997dc4cea3275e5ee1a1f5e2993627be

I'm just directly using the existing User reference instead of creating a new reference.

Reply:
I get why you are doing this, however please add a new type alias for `resourceReference` like the rest of PagerDuty resource are doing.

Reply:
@imjaroiswebdev Sure I can do that, but just for my own understanding, what's the reasoning for adding a new alias when we'd like to just use a user reference in this case?

Reply:
I'm assuming I would add the following to [`references.go`](https://github.com/heimweh/go-pagerduty/blob/master/pagerduty/references.go)
```golang
// EventOrchestrationCacheVariableReference represents a reference to an
// Event Orchestration Cache Variable schema
type EventOrchestrationCacheVariableReference resourceReference
```

However, this would not be accurate, since the `CreatedBy` and `UpdatedBy` fields (i.e. the fields using this reference) would be `UserReference`s, no?

Reply:
Hmmm... now I get what You're saying @c-kaieong, You are right, sorry about the mixup, `CreatedBy` and `UpdatedBy` are indeed `UserReference`s as you are noting, so a new `EventOrchestrationCacheVariableReference` is not needed here.